docs/ruleset_functions/get_SWH_equipment_associated_with_each_SWH_bat.md

```diff
@@ -0,0 +1,60 @@
+
+Returns:
+- **swh_and_equip_dict**: A dictionary containing where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 in the RMD and values are dictionaries where keys are the type of SWH equipment and values are the ids of the connected equipment.  Example:  
+{"DORMITORY":{"SWHDistribution":["swhd1","swhd2"], "SWHHeatingEq":["swh_eq1","swh_eq2"], "Pumps":["p1"], "Tanks":["t1"], "Piping":["piping1"], "SolarThermal":[], "SWH_Uses":["swh_use1","swh_use2"], "EnergyRequired": 100000}}
+
+Function Call:
+
+- get_spaces_associated_with_each_SWH_bat  
+- get_component_by_ID  
+- get_all_child_SWH_piping_ids  
+- get_building_segment_swh_bat  
+- get_energy_required_to_heat_swh_use  
+- get_SWH_uses_associated_with_each_building_segment  
+
+Data Lookup: None
+
+Logic:
+
+- create a blank dictionary: `swh_and_equip_dict = {}`
+- look at each building segment in the RMD: `for building_segment in RMD...building_segments:`
+    - get the SWH BAT for this building segment using the function get_building_segment_swh_bat: `swh_bat = get_building_segment_swh_bat(RMD,building_segment)`
+    - create the blank list for the swh_bat, if it doesn't exist yet: `swh_and_equip_dict[swh_bat].set_default({"SWHDistribution":[],"SWHHeatingEq":[],"Pumps":[],"Tanks":[], "Piping":[], "SolarThermal":[], "SWH_Uses":[], "EnergyRequired":0})`
+    - get the service water heating uses in the building segment `service_water_heating_use_ids = get_SWH_uses_associated_with_each_building_segment(RMD, building_segment.id)`
+    - look at each service water heating use id: `for swh_use_id in service_water_heating_use_ids:`
+        - get the swh_use using get_component_by_ID: `swh_use = get_component_by_ID(RMD, swh_use_id)`
+        - get the energy required to heat the water using the function get_energy_required_to_heat_swh_use: `energy_required = get_energy_required_to_heat_swh_use(swh_use, RMD)`
+        - add the energy required to heat the swh use, to "EnergyRequired" for this bat: `swh_and_equip_dict[swh_bat]["EnergyRequired"] += energy_required`
+        - add the id for the swh_use to the "SWH_Uses" list: `swh_and_equip_dict[swh_bat]["SWH_Uses"].append(swh_use.id)`
+        - get the distribution id: `distribution_id = swh_use.served_by_distribution_system`
+        - check if the distribution system is already in the list.  If it is not we'll add it and all associated equipment: `if !distribution_id in? swh_and_equip_dict[swh_bat]["SWHDistribution"]:`
+            - add the distribution id to the list: `swh_and_equip_dict[swh_bat]["SWHDistribution"].append(distribution_id)`
+            - get the distribution using get_component_by_ID: `distribution = get_component_by_ID(RMD, distribution_id)`
+            - now we need to get all of the equipment connected to this distribution system.
+            - get the tanks: `tanks = distribution.tanks`
+            - append the tank ids to the dictionary: `for t in tanks:  swh_and_equip_dict[swh_bat]["Tanks"].append(t.id)`
```

Comment:
This is a list of tanks so I think this logic needs to be updated since it is adding a single id